### PR TITLE
[R] various bug fixes and enhancements

### PR DESCRIPTION
**Pull request by jokedurnez**
### PR checklist

- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

I cleaned up the mustache files to be more readable, fixed the primary types and added conditionals to responses. 

Used for this specification: https://github.com/openlattice/api/blob/develop/openlattice.yaml
Thanks for the great resource !

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/r/api.mustache [hunk 1]
```diff
@@ -95,10 +95,15 @@
...
-        returnObject <- {{returnType}}$new()
-        result <- returnObject$fromJSON(httr::content(resp, "text", encoding = "UTF-8"))
-        Response$new(returnObject, resp)
-        {{/returnType}}
+          {{#isPrimitiveType}}
+                returnObject <- {{returnType}}$new()
+                result <- returnObject$fromJSON(httr::content(resp, "text", encoding = "UTF-8"))
+                Response$new(returnObject, resp)
+           {{/isPrimitiveType}}
+          {{^isPrimitiveType}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/r/api.mustache]: @jokedurnez thanks for the PR. Seems like this line is missing a space in terms of alignment?

**Comment by wing328** [comment]
@jokedurnez thanks for the enhancement. Can you please also run the following to update the R Petstore samples?

linux: `./bin/r-petstore.sh`
windows: `bin\windows\r-petstore.bat`

(you may notice changes not caused by this PR, which is fine)

**Comment by jokedurnez** [comment]
Fixed those review comments !

**Comment by wing328** [comment]
@jokedurnez shall we use 2-space for indentation as stated in https://google.github.io/styleguide/Rguide.xml#indentation ?

**Comment by wing328** [comment]
Also I've started a project to refactor the R client generator and you can track progress in https://github.com/OpenAPITools/openapi-generator/projects/8#card-17476252

**Comment by wing328** [comment]
FYI. Filed https://github.com/OpenAPITools/openapi-generator/pull/2215 to refactor the R client.

File: modules/openapi-generator/src/main/resources/r/model.mustache [hunk 2]
```diff
@@ -74,63 +90,94 @@
...
+            {{#isPrimitiveType}}
+                self$`{{baseName}}`{{#hasMore}},{{/hasMore}}
+            {{/isPrimitiveType}}
+            {{^isPrimitiveType}}
+                self$`{{baseName}}`$toJSON(){{#hasMore}},{{/hasMore}}
+            {{/isPrimitiveType}}
         {{/isListContainer}}
         {{/vars}}
       )
+      gsub("[\r\n]| ", "", outstring)
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/r/model.mustache]: I think this may cause issues if the value contains these characters

**Pull request merged**